parser/impl/parser.cpp

```diff
@@ -240,28 +240,123 @@ sPtr<ast::RecordType> Parser::parseRecordType() {
-            .pos = token.pos,
-            .message = "Expected \"var\" keyword but didn't find it.",
-        });
+  
+    sPtr<ast::Variable> Parser::parseVariable() {
+	    ast::Variable variable;
+        Token token = skipWhile(isNewLine);
+        if (token.type != TokenType::Var) {
+            m_errors.push_back(Error{
+                .pos = token.pos,
+                .message = "Expected \"var\" keyword but didn't find it.",
+            });
+        }
+        token = skipWhile(isNewLine);
+        if (token.type != TokenType::Identifier) {
+            m_errors.push_back(Error{
+                .pos = token.pos,
+                .message = "Expected an identifier after \"var\" keyword.",
+            });
+        }
+	    variable.name = token;
+        token = skipWhile(isNewLine);
+        if (token.type != TokenType::Is && token.type != TokenType::Colon) {
+            m_errors.push_back(Error{
+                .pos = token.pos,
+                .message = "Expected an \"is\" keyword or : after the identifier.",
+            });
+        }
+        switch (token.type)
+        {
+	        case TokenType::Colon:
+                variable.type = parseType();
+                if ((token = m_lexer.Peek()).type != TokenType::Is) break;
+		        else m_lexer.Next();
+            case TokenType::Is:
+                variable.expression = parseExpression();
+                break;
+            default:
+                m_errors.push_back(Error{
+                    .pos = token.pos,
+                    .message = "Expected an \"is\" keyword or : after the identifier.",
+                });
+        }
+        return std::make_shared<ast::Variable>(variable);
+  }        
+  
+    sPtr<ast::Body> Parser::parseBody() {
+	    ast::Body body;
+        Token currentToken;
+        while ((currentToken = m_lexer.Peek()).type != TokenType::End)
+        {
+            switch (currentToken.type)
+            {
+            case TokenType::Var:
+                body.variables.push_back(parseVariable());
+                continue;
+            case TokenType::Type:
+                body.types.push_back(parseType());
+                continue;
+            case TokenType::NewLine:
+                m_lexer.Next();
+                continue;
+            default:
+                body.statements.push_back(parseStatement());
+            }
+        }
+        return std::make_shared<ast::Body>(body);
+    }
+
+    sPtr<ast::Statement> Parser::parseStatement() {
+        auto currentToken = skipWhile(isNewLine);
+        sPtr<ast::Expression> expression;
+        switch (currentToken.type)
+        {
+        case TokenType::Identifier:{
+	        expression = parseExpression();
+            ast::RoutineCall* routineCall = dynamic_cast<ast::RoutineCall*>(expression.get());
+	        if (routineCall != nullptr)
+	            return std::make_shared<ast::RoutineCall>(*routineCall);
+            else 
+                return parseAssignment(expression);
```

Comment:
[Discussion] I think it will be simpler if we create a common "Identifier" node which is to be replaced with either `RoutineCall` or `Variable` in a later pass depending on the defined routine/variable names and scopes